code/modules/xenomorph/xeno_structures.dm

```diff
@@ -1175,3 +1175,238 @@ TUNNEL
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.adjust_stagger(3)
+			H.apply_damage(30, BRUTE, "chest", H.get_soft_armor("melee", "chest"))
+		qdel(src)
+		return TRUE
+
+	to_chat(user, span_xenowarning("We shed our shattered scales as new ones grow to replace them!"))
+	var/mob/living/carbon/xenomorph/X = user
+	X.adjust_sunder(-sunder_removal)
+	qdel(src)
+	return TRUE
+
+/obj/structure/xeno/plant/plasma_fruit
+	name = "power fruit"
+	desc = "A cyan fruit, beating like a creature's heart"
+	icon_state = "plasma_fruit_immature"
+	mature_icon_state = "plasma_fruit"
+	///How much bonus plasma should we restore during the duration, 1 being 100% from base regen
+	var/bonus_regen = 1
+	///How long should the buff last
+	var/duration = 1 MINUTES
+
+/obj/structure/xeno/plant/plasma_fruit/on_use(mob/user)
+	to_chat(user, span_warning("We begin consuming [src]..."))
+	if(!do_after(user, 2 SECONDS, FALSE, src))
+		return FALSE
+	if(!isxeno(user))
+		visible_message(span_warning("[src] releases a sticky substance before spontaneously bursting into flames!"))
+		flame_radius(3, get_turf(src), colour = "green")
+		qdel(src)
+		return TRUE
+
+	var/mob/living/carbon/xenomorph/X = user
+	if(isxenoravager(X)) //Ask if this should be made into a trait for xenos with special ressources
+		to_chat(X, span_xenowarning("But our body rejects the fruit, our fury does not build up with a healthy diet!"))
+		return FALSE
+	X.apply_status_effect(/datum/status_effect/plasma_surge, X.xeno_caste.plasma_max, bonus_regen, duration)
+	to_chat(X, span_xenowarning("[src] Restores our plasma reserves, our organism is on overdrive!"))
+	qdel(src)
+	return TRUE
+
+
+/obj/structure/xeno/plant/stealth_plant
+	name = "night shade"
+	desc = "A beautiful flower, what purpose it could serve to the alien hive is beyond you however..."
+	icon_state = "stealth_plant_immature"
+	mature_icon_state = "stealth_plant"
+	maturation_time = 5 MINUTES
+	///The radius of the passive structure camouflage, requires line of sight
+	var/camouflage_range = 5
+	///The range of the active stealth ability, does not require line of sight
+	var/active_camouflage_pulse_range = 10
+	///How long should veil last
+	var/active_camouflage_duration = 20 SECONDS
+	///How long until the plant can be activated again
+	var/cooldown = 2 MINUTES
+	///Is the active ability veil on cooldown ?
+	var/on_cooldown = FALSE
+	///The list of passively camouflaged structures
+	var/list/obj/structure/xeno/camouflaged_structures = list()
+	////The list of actively camouflaged xenos by veil
+	var/list/mob/living/carbon/xenomorph/camouflaged_xenos = list()
+
+/obj/structure/xeno/plant/stealth_plant/Initialize()
+	. = ..()
+	START_PROCESSING(SSslowprocess, src)
+
+/obj/structure/xeno/plant/stealth_plant/Destroy()
+	for(var/obj/S in camouflaged_structures)
+		S.alpha = 255
+	unveil()
+	STOP_PROCESSING(SSslowprocess, src)
+	return ..()
+
+/obj/structure/xeno/plant/stealth_plant/process()
+	if(!mature)
+		return
```

Comment:
does it stop processing? I don't get what's going on

Reply:
Mistake from me, I made it start processing on init and check for mature each time. I'll make it start processing on mature.

Reply:
Stops processing on Destroy otherwise *